charm.go

```diff
@@ -17,10 +17,13 @@ var logger = loggo.GetLogger("juju.charm")
 // may be handled as a charm.
 type Charm interface {
 	Meta() *Meta
+	BasesManifest() *Manifest
 	Config() *Config
 	Metrics() *Metrics
 	Actions() *Actions
 	Revision() int
+	Format() Format
```

Comment:
Format shouldn't leak outside of `juju/charm`. Instead, we should ask the library what things it has based on the charm. An example of this would be `ComputedSeries`...

```go
func ComputedSeries(charm Charm) []string {
    ...
}
```

That would remove the need to constantly add methods to charm interface.